config/initializers/warden_hooks.rb

```diff
@@ -0,0 +1,9 @@
+Warden::Manager.after_set_user do |user, auth, opts|
+  scope = opts[:scope]
+  auth.cookies.signed["#{scope}.id"] = user.id
```

Comment:
Why do we have to handle cookies ourselves? We have access to the id of the currently logged in user via `current_user.id`.

Reply:
I'll just quote the tutorial we followed here.

> When establishing a websocket connection, we don’t have access to the user session but we do have access to the cookies. So, in order to be able to authenticate the user we need to do some devise related stuff first.

It is used in conjunction with `app/channels/application_cable/connection.rb#find_verified_user`

Reply:
I added the above explanation as a comment.

Reply:
Thanks for the explanation :+1: